pkg/controller/topology.go

```diff
@@ -157,12 +160,15 @@ func GenerateAccessibilityRequirements(
 	)
 
 	// 1. Get CSINode for the selected node
-	if selectedNode != nil {
-		selectedCSINode, err = getSelectedCSINode(csiNodeLister, selectedNode)
+	var topologyKeys []string
+	if len(selectedNodeName) > 0 {
```

Comment:
This logic is very hard to follow with the cache lookups and updates being interspersed with condition checks.  I think it would be easier to read if we wrapped the informer/cache lookup logic, and kept the core function clean with just business logic. Something like:

```
func topologyLookup(selectedNodeName) {

csiNode, err1 := getSelectedCSINode(csiNodeLister, selectedNodeName)
node,err2 := getNode(nodeLister, selectedNodeName)

if csiNode == nil || node  == nil {
  topologyInfo = cache.getTopologyInfo(pvc)
} else {
  topologyInfo = { getTopologyKeys(csiNode, getNodeLabels(node, topologyKeys)}
  cache.addToplogyInfo(topologyInfo)
}

return topologyInfo
}
```